File: climb-log-flutter/lib/climasys/api/models/log_climb_request.dart
// models/route.dart
class LogClimbRequest {
  final int routeId;
  final int? difficultyForGradeId;
  final int? attemptCount;
  final String? notes;

  LogClimbRequest({
    required this.routeId,
    required this.difficultyForGradeId,
    required this.attemptCount,
    required this.notes
  });

  Map<String, dynamic> toJson() => {
    'routeId': routeId,
    'difficultyForGradeId': difficultyForGradeId,
    'attemptCount': attemptCount,
    'notes': notes
  };
}

File: ClimbLog.Api/ClimbLog.Application/Features/ClimbLog/Commands/LogClimb/LogClimbRequest.cs

namespace ClimbLog.Application.Features.ClimbLog.Commands.LogClimb;
public class LogClimbRequest
{
    public int RouteId { get; set; }
    public int? DifficultyForGradeId { get; set; }
    public int? AttemptCount { get; set; }
    public string? Notes { get; set; }
}
